src/bridge/test-httpstream.c

```diff
@@ -833,8 +828,6 @@ test_tls_authority_bad (TestTls *test,
                           "options", options,
                           NULL);
 
-  cockpit_expect_log ("cockpit-bridge", G_LOG_LEVEL_MESSAGE,
-                      "*Unacceptable TLS certificate:*untrusted-issuer*");
   cockpit_expect_log ("cockpit-bridge", G_LOG_LEVEL_MESSAGE,
                       "*Unacceptable TLS certificate");
```

Comment:
This is a bit thin now, though. Does GSocketClient really not raise an error or at least write a warning with something more specific than "Unacceptable cert"? It doesn't need to  be machine parseable, but something like "expired" vs. "unknown CA" would surely be helpful for debugging your system?

Reply:
I was also surprised to discover that it doesn't give a better error message.  And this is really deep inside of GSocketClient, so it's not super easy to plumb it up.  This is the basic reason we augmented the messages ourselves, I think... Ideally, glib-networking just needs to give better errors on certificate rejections in the first place...

The code in question in glib-networking, by the way:

  if (ret == GNUTLS_E_CERTIFICATE_ERROR)
    {
      g_clear_error (&my_error);
      g_set_error_literal (error, G_TLS_ERROR, G_TLS_ERROR_BAD_CERTIFICATE,
                           _("Unacceptable TLS certificate"));
      return G_TLS_CONNECTION_BASE_ERROR;
    }

Reply:
Would you be happy with an upstream report asking for better error reporting?

Reply:
Yes, I'm happy with an upstream report for that.